packages/react/src/rise.tsx

```diff
@@ -198,10 +199,17 @@ export function BaseRise({
           )
         }
       }
-      return (
-        <Component {...componentProps}>
-          {render(stateNode.children, [...path, 'children'])}
-        </Component>
+      // @ts-ignore
+      const jsxFactory = stateNode.static ? jsxs : jsx
```

Comment:
Note: [React uses same factory in production for both static and dynamic](https://github.com/facebook/react/blob/main/packages/react/src/jsx/ReactJSX.js#L20-L21). They only differentiate during development, likely to provide that key warning among other improvements.

It's probably better for us to still call static or dynamic selectively, in case that behavior changes in the future.

However, since same function is used in production right now, it leads to conclusion that differences if any are mostly dev specific.

Reply:
Alternatively, we could always call `jsxs`, no matter whether children are static or dynamic and that would trigger no warning on the client about missing keys. We would not add any additional properties to ComponentDataState.

This might be beneficial, because if you're using Rise Playground, you won't be able to see the logs and key warnings should appear on the server instead!

The only difference is during development when using this in-house as a part of your app. In that case you have access to logging and you should be able (or maybe not?) to see that!